### DISCUSSION TITLE
How do I extend an existing workflow for different trigger?

### DISCUSSION THREAD

**Discussion by rdgoite** [community/community#25322] [Actions]
<p>I have 1 workflow that is triggered by 2 different events. However, it’s come to a point where I need to extend it for one of the triggers; it would still do the same thing, but for one of the triggers, it would perform some additional steps right after. How do I achieve something like this? I know I can just copy the workflow to a new file with the extended steps, but it’d be nice if I can reuse some part of a workflow spec without have to copy the actual spec.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That sounds like something you can do with <code>if:</code> conditions either on steps or jobs using data from the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><code>github</code> context</a>. If you just want to know about the event type you can use <code>github.event_name</code>, if you need event details you’ll have to look into <code>github.event</code> properties.</p>

**Comment by rdgoite** [comment]
<p>Ah! I didn’t know there’s an <code>if</code> conditional. That’s actually the first thing that came to mind but I didn’t bother to check for it because I thought it’d be weird for the workflow spec to have it because it’s a spec and it’s in YAML. I guess another lesson here is verify assumptions.</p>
<p>I think this could work. Thanks!</p>
